src/scripts/keyboard.js

```diff
@@ -0,0 +1,561 @@
+      key.classList.toggle('keyboard__key--active', this.properties.shift);
+    });
+  },
+
+  _toggleSound() {
+    this.properties.sounds = !this.properties.sounds;
+    document.querySelector('.keyboard__key--volume').innerHTML = this.properties.sounds ? '<i class="material-icons">volume_up</i>' : '<i class="material-icons">volume_off</i>';
+  },
+
+  _toggleLang() {
+    document.querySelector('.keyboard').remove();
+    this.properties.lang = this.properties.lang === 'en' ? this.properties.lang = 'ru' : 'en';
+    this.init();
+    this.elements.main.classList.remove('keyboard--hidden');
+  },
+
+  open(initialValue, oninput, onclose) {
+    this.properties.value = initialValue || '';
+    this.eventHandlers.oninput = oninput;
+    this.eventHandlers.onclose = onclose;
+    this.elements.main.classList.toggle('keyboard--hidden');
+    this.properties.cursorPosition = document.querySelector('.use-keyboard-input').selectionStart;
+  },
+
+  close() {
+    this.properties.value = '';
+    this.eventHandlers.oninput = oninput;
+    this.eventHandlers.onclose = onclose;
+    this.elements.main.classList.add('keyboard--hidden');
+  },
+
+  playClick() {
+    const shiftClick = new Audio();
+    const backspaceClick = new Audio();
+    const capsClick = new Audio();
+    const enterClick = new Audio();
+    const typeClick = new Audio();
+    const closeClick = new Audio();
+    const cursorMoveClick = new Audio();
+    const speechClick = new Audio();
+
+    shiftClick.volume = this.properties.soundVolume;
+    backspaceClick.volume = this.properties.soundVolume;
+    capsClick.volume = this.properties.soundVolume;
+    enterClick.volume = this.properties.soundVolume;
+    typeClick.volume = this.properties.soundVolume;
+    closeClick.volume = this.properties.soundVolume;
+    cursorMoveClick.volume = this.properties.soundVolume;
+    speechClick.volume = this.properties.soundVolume;
+
+    shiftClick.preload = 'auto';
+    backspaceClick.preload = 'auto';
+    capsClick.preload = 'auto';
+    enterClick.preload = 'auto';
+    typeClick.preload = 'auto';
+    closeClick.preload = 'auto';
+    cursorMoveClick.preload = 'auto';
+    speechClick.preload = 'auto';
+
+    shiftClick.src = this.soundsSets[this.properties.lang].shift;
+    backspaceClick.src = this.soundsSets[this.properties.lang].backspace;
+    capsClick.src = this.soundsSets[this.properties.lang].caps;
+    enterClick.src = this.soundsSets[this.properties.lang].enter;
+    typeClick.src = this.soundsSets[this.properties.lang].type;
+    closeClick.src = this.soundsSets.closeSound;
+    cursorMoveClick.src = this.soundsSets.cursorMoveSound;
+    speechClick.src = this.soundsSets.speechSound;
+
+    const shiftBtn = document.querySelectorAll('.keyboard__key--shift');
+    shiftBtn.forEach((elem) => {
+      elem.addEventListener('click', () => {
+        if (this.properties.sounds) shiftClick.play();
+      });
+    });
```

Comment:
This block should be converted to a function. You have 7 repetitive code blocks.